response_operations_ui/contexts/case.py

```diff
@@ -23,25 +23,36 @@ def build_response_status_context(
                 case_group_id=case_group_id,
                 period=ce_period,
             ),
-            "radios": _generate_radios(allowed_transitions_for_case, case_completed_time),
-            "cancel_link": url_for("reporting_unit_bp.view_reporting_unit_survey", ru_ref=ru_ref, survey_id=survey_id),
+            "radios_section": _generate_radios_section(
+                allowed_transitions_for_case, case_completed_time, ru_ref, survey_id
+            ),
         }
     else:
         change_response_status = {}
     return {"change_response_status": change_response_status}
 
 
-def _generate_radios(allowed_transitions_for_case: dict, case_completed_time: datetime) -> list:
+def _generate_radios_section(
+    allowed_transitions_for_case: dict, case_completed_time: datetime, ru_ref: str, survey_id: str
+) -> dict:
     radios = []
+    enabled_radios = True
     for index, (event, status) in enumerate(allowed_transitions_for_case.items()):
         radio = {"id": f"state-{index + 1}", "label": {"text": status}, "value": event}
         if event == "COMPLETED_TO_NOTSTARTED" and case_completed_time:
+            enabled_radios = False
```

Comment:
I'd move this boolean to something like:

 ```
       if event == "COMPLETED_TO_NOTSTARTED" and case_completed_time:
            complete_to_not_started_wait_time = app.config["COMPLETE_TO_NOT_STARTED_WAIT_TIME"]
            if (datetime.now() - case_completed_time) < timedelta(seconds=complete_to_not_started_wait_time):
                radio["attributes"] = {"disabled": "true"}
                radio["label"][
                    "description"
                ] = "Status can only be changed after 48 hours have passed since the submission"
                enabled_radios = False
            else:
                enabled_radios = True
```

Reply:
I think I may leave it as is because if we add another state change it may not require the 48 grace period so if one radio is enabled we want the button to be enabled - and if the disabled one is processed second in the loop, the button will remain disabled while a transition can be made

Reply:
Ah okay. That makes.